sys/net/network_layer/fib/fib.c

```diff
@@ -421,6 +421,41 @@ int fib_get_next_hop(kernel_pid_t *iface_id,
     return 0;
 }
 
+int fib_get_destination_set(uint32_t bitmap, uint32_t type,
+                            fib_destination_set_entry_t *dst_set, size_t* dst_set_size)
+{
+    mutex_lock(&mtx_access);
+    int ret = -EHOSTUNREACH;
+    size_t found_entries = 0;
+
+    for (size_t i = 0; i < FIB_MAX_FIB_TABLE_ENTRIES; ++i) {
+        if( (fib_table[i].global_flags & bitmap) == (type & bitmap) ) {
+
+            if( dst_set != NULL && found_entries < (*dst_set_size)-1 ) {
```

Comment:
I see. Actually I thought that this function should return an error when no `dst_set` was specified. Anyhow, can you extend the doc that this function still counts and returns all matching entries even when they are not copied to `dst_set`?

And add parens to the condition, please (: